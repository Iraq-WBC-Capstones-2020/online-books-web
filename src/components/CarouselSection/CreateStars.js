import React from 'react';
import PropTypes from 'prop-types';
import { round } from 'mathjs';
function CreateStars({ rate, starSize }) {
  const stars = [];
  rate = round(rate, 1);
  const firstDigit = round(rate - parseInt(rate), 1);
  let secondDigit = firstDigit.toString().split('.')[1];
  const Size =
    starSize === 'small'
      ? 'fa-sm lightBlue mrSmallStars'
      : 'fa-md mr-1 text-yellow-300';

  let createStars = () => {
    for (let i = 1; i <= 5; i++) {
      if (i <= rate) {
        stars.push(<icon className={`fa fa-star ${Size}`} key={i}></icon>);
      } else {
        if (secondDigit >= 5) {
          secondDigit = 0;
          stars.push(
            <icon className={`fas fa-star-half-alt ${Size}`} key={i}></icon>
          );
        } else {
          stars.push(
            <icon className={` fa fa-star font-light ${Size}`} key={i}></icon>
          );
        }
      }
    }
    return stars;
  };
  return <>{createStars()}</>;
}
CreateStars.propTypes = {
  rate: PropTypes.number,
};
export default CreateStars;
