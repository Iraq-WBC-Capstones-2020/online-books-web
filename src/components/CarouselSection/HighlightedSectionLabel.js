import React from 'react';
import PropTypes from 'prop-types';

function HighlightedSectionLabel(props) {
  return (
    <div className="buttomText bg-black  pt-1 pt-md-2 text-white font-bold text-center  highlight ">
      {props.text}
    </div>
  );
}
HighlightedSectionLabel.propTypes = {
  text: PropTypes.string,
};
export default HighlightedSectionLabel;
