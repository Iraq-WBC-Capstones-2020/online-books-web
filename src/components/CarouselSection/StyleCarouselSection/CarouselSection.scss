@import '../../../assets/main.css';

$gray-color: #979797;
$pink-color: #eec197;
$light-blue: #8bb4ea;
$dark-blue: #7aa2d6;
$light-gray: #f6f6f6;
$light-blue2: #f9f9f9;
$font-roboto: 'Roboto';
$light-pink: #fee4cb;

.mycolorPink {
  background: $pink-color;
}

%fontType {
  font-family: 'Saira';
}

.fontHeader1 {
  @extend %fontType;
  color: $dark-blue;
}

.fontHeader2 {
  @extend %fontType;
  color: $gray-color;
  font-size: 1.1rem;
}

.parg1 {
  color: $gray-color;
  font-family: $font-roboto;
  font-size: 0.9rem;
}

.fontPrice {
  font-family: $font-roboto;
  color: #333333;
  font-size: 1.5rem;
}

.buttomText {
  font-family: $font-roboto;
  font-size: 1rem;
}

.highlight {
  font-family: $font-roboto;
  width: 15rem;
  height: 2.3rem;
  @media (max-width: 640px) {
    width: auto;
    height: 2rem;
  }
}

.bgGray {
  background: $light-gray;
}

.bgLightblue {
  background: $light-blue2;
}

.lightblue {
  background: $light-blue;
}

.imgGrideCarsouel {
  display: block;
  width: 100%;
  margin: 0 auto;
  height: 9rem;
  @media (max-width: 640px) {
    height: 5rem;
  }
}

.imageCardCarsouel {
  display: block;
  width: 100%;
  height: 19rem;
}

.active-card {
  left: 0px !important;
  position: relative !important;
}

.inactive-card {
  left: +100% !important;
  position: absolute !important;
}

.Card{
  border-style: none !important;
}
