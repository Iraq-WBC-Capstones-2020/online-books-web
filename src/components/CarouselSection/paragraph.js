import React from 'react';
import { useTranslation } from 'react-i18next';

function Paragraph() {
  const { t } = useTranslation();
  return (
    <p className="font-serif text-center text-blue-600 font-bold my-12 text-3xl bgLightblue">
      {t('mainPage.text')}
    </p>
  );
}
export default Paragraph;
