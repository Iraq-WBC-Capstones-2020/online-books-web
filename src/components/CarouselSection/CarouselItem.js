import React from 'react';
import PropTypes from 'prop-types';
import {
  MDBCard,
  MDBCardBody,
  MDBCardText,
  MDBCarouselItem,
  MDBCol,
  MDBRow,
  MDBContainer,
  MDBIcon,
  MDBBtn,
} from 'mdbreact';
import { useTranslation } from 'react-i18next';
import CreateStars from './CreateStars';

// create 4 items for the carsouel movement
function CarouselItems({ activeCard, handelChangeCard }) {
  const { t } = useTranslation();

  const booksArray = [
    {
      cover:
        'http://1.bp.blogspot.com/-yQklGusIOBE/Ul-oQ0_9vOI/AAAAAAAAfI4/XQTmNLaJZLQ/s1600/book-cover-poster-27.jpg',
      authorName: 'Alice Mayer',
      bookName: 'The Shadows: A Novel',
      type: 'formate:Hardback',
      price: '$31.5',
      rate: 3.27,
      views: '22,333',
      id: 1,
    },
    {
      cover:
        'http://www.snugglyoranges.com/wp-content/uploads/2014/09/Unspoken.jpg',
      authorName: 'Alice Mayer',
      bookName: 'The Shadows: A Novel',
      type: 'formate:Hardback',
      price: '$31.5',
      rate: 3.5,
      views: '22k',
      id: 2,
    },
    {
      cover:
        'https://s-media-cache-ak0.pinimg.com/736x/78/c5/f0/78c5f0de85565a0fd4a911f2681f56e3--book-cover-art-cover-books.jpg',
      authorName: 'Alice Mayer',
      bookName: 'The Shadows: A Novel',
      type: 'formate:Hardback',
      price: '$31.5',
      rate: 3.76,
      views: '22k',
      id: 3,
    },
    {
      cover:
        'https://th.bing.com/th/id/OIP.Zf2SkeIiSI4ymvT2cc4u2AHaKd?pid=Api&rs=1',
      authorName: 'Alice Mayer',
      bookName: 'The Shadows: A Novel',
      type: 'formate:Hardback',
      price: '$31.5',
      rate: 3,
      views: '22k',
      id: 4,
    },
    {
      cover:
        'https://s-media-cache-ak0.pinimg.com/736x/78/c5/f0/78c5f0de85565a0fd4a911f2681f56e3--book-cover-art-cover-books.jpg',
      authorName: 'Alice Mayer',
      bookName: 'The Shadows: A Novel',
      type: 'formate:Hardback',
      price: '$31.5',
      rate: 3.73,
      views: '22k',
      id: 5,
    },
    {
      cover:
        'https://s-media-cache-ak0.pinimg.com/736x/78/c5/f0/78c5f0de85565a0fd4a911f2681f56e3--book-cover-art-cover-books.jpg',
      authorName: 'Alice Mayer',
      bookName: 'The Shadows: A Novel',
      type: 'formate:Hardback',
      price: '$31.5',
      rate: 3.46,
      views: '22k',
      id: 6,
    },
  ];

  return (
    <div>
      {booksArray.slice(0, 4).map((book, index) => (
        <MDBCarouselItem
          itemId={index}
          key={book.id}
          className={`
            ${activeCard === index ? 'active-card' : 'inactive-card'}
             mt-md-4 mb-md-4 pl-lg-3`}
        >
          <MDBCard className="w-30 shadow-none" wide>
            <MDBContainer className="w-full">
              <MDBRow className="p-0">
                <MDBCol size="12" md="4" lg="5" className="p-0">
                  <img
                    className="imageCardCarsouel"
                    src={`${book.cover}`}
                    alt="cover book"
                  />
                </MDBCol>

                <MDBCol md="1" lg="1"></MDBCol>

                <MDBCol md="7" lg="6" size="12" className="mb-0 mt-auto">
                  <MDBCardBody className="p-lg-0 text-center text-md-left text-lg-left mx-auto mt-md-2">
                    <div>
                      <h1 className="fontHeader1 mb-0 font-weight-normal  text-2xl">
                        {book.authorName}
                      </h1>
                    </div>
                    <MDBCardText>
                      <p className="mb-1 fontHeader2 font-normal font-weight-normal ">{`${book.bookName}`}</p>
                      <p className="mb-1 parg1 font-weight-normal">
                        {book.type}
                      </p>
                      <p className="font-black fontPrice font-weight-normal ">{`${book.price}`}</p>

                      <div className="mb-3">
                        <div className="inline-block mr-2">
                          <CreateStars rate={book.rate} />
                          {'  '}
                          {book.views} views
                        </div>
                      </div>
                    </MDBCardText>

                    <button className="buttomText text-center mycolorPink w-5/6 shadow-md  hover:shadow-lg h-10 text-black font-bold leading-3 rounded-full">
                      {t('addtocart')}
                    </button>
                  </MDBCardBody>
                  {/*arrow button// d-flex justify-content-end */}
                  <div className="h-16 relative">
                    <div className="absolute bottom-0 right-0">
                      <MDBBtn
                        color=""
                        className="shadow-sm p-0 h-8"
                        onClick={() => handelChangeCard(true)}
                      >
                        <MDBIcon
                          icon="chevron-left"
                          size="lg"
                          className="px-2 py-1"
                        />
                      </MDBBtn>
                      <MDBBtn
                        color=""
                        className="shadow-sm p-0 h-8"
                        onClick={() => handelChangeCard(false)}
                      >
                        <MDBIcon
                          icon="chevron-right"
                          size="lg"
                          className="px-2 py-1"
                        />
                      </MDBBtn>
                    </div>
                  </div>
                </MDBCol>
              </MDBRow>
            </MDBContainer>
          </MDBCard>
        </MDBCarouselItem>
      ))}
    </div>
  );
}
CarouselItems.propTypes = {
  activeCard: PropTypes.number,
  handelChangeCard: PropTypes.func,
};
export default CarouselItems;
