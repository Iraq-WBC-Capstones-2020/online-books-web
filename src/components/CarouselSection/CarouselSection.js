import React, { useState, useEffect, useRef } from 'react';
import {
  MDBCarousel,
  MDBCarouselInner,
  MDBCol,
  MDBRow,
  MDBContainer,
} from 'mdbreact';
import './StyleCarouselSection/CarouselSection.scss';
import CarouselItems from './CarouselItem';
import GrideCarousel from './GridCarousel';
import HighlightedSectionLabel from './HighlightedSectionLabel';

function CarsouelSection({highlightText}) {
  const [activeCardItem, setActiveCardItemState] = useState(0);
  const timerId = useRef();
  const firstCardIndex = 0;
  const lastCardIndex = 3;

  useEffect(() => (timerId.current = resetTimer()), []);
  let resetTimer = () => {
    return setTimeout(() => handelChangeCard(true, 'unClicked'), 5000);
  };

  let handelChangeCard = (increase, button) => {
    if (increase) {
      button === 'unClicked'
        ? setActiveCardItemState((oldCard) =>
            oldCard === lastCardIndex ? firstCardIndex : oldCard + 1
          )
        : setActiveCardItemState((oldCard) =>
            oldCard === lastCardIndex ? oldCard : oldCard + 1
          );
    } else {
      setActiveCardItemState((oldCard) =>
        oldCard === firstCardIndex ? oldCard : oldCard - 1
      );
    }
    clearTimeout(timerId.current);
    timerId.current = resetTimer();
  };
  return (
    <>
      <MDBContainer>
        <MDBRow center>
          <MDBCol md="11" lg="10" xl="8">
            <HighlightedSectionLabel text={highlightText} />

            <MDBContainer fluid className="bgGray ">
              <MDBRow>
                <MDBCol md="12" lg="8" xl="8">
                  <MDBCarousel
                    activeItem={activeCardItem}
                    length={4}
                    showControls={false}
                    showIndicators={false}
                    slide
                  >
                    <MDBCarouselInner>
                      <CarouselItems
                        handelChangeCard={handelChangeCard}
                        activeCard={activeCardItem}
                      />
                    </MDBCarouselInner>
                  </MDBCarousel>
                </MDBCol>

                <MDBCol
                  xs="12"
                  sm="12"
                  md="12"
                  lg="4"
                  xl="4"
                  className="mt-3 mb-md-0 mt-md-0 mb-lg-4 mt-lg-4  "
                >
                  {/* gride for the new release and best sellers  */}
                  <GrideCarousel activeCardItem={activeCardItem} />
                </MDBCol>
              </MDBRow>
            </MDBContainer>
          </MDBCol>
        </MDBRow>
      </MDBContainer>
    </>
  );
}

export default CarsouelSection;
