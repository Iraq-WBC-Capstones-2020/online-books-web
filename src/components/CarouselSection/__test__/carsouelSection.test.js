import React from 'react';
import renderer from 'react-test-renderer';
import CarouselSection from '../CarouselSection';

test('CarouselSection snapshot', () => {
  const tree = renderer.create(<CarouselSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
