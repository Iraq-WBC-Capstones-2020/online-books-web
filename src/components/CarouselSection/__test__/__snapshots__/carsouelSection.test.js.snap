// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSection snapshot 1`] = `
<div
  className="container"
  data-test="container"
>
  <div
    className="row justify-content-center"
    data-test="row"
  >
    <div
      className="col-md-11 col-lg-10 col-xl-8"
      data-test="col"
    >
      <div
        className="buttomText bg-black  pt-1 pt-md-2 text-white font-bold text-center  highlight "
      >
        best sellers in books
      </div>
      <div
        className="container-fluid bgGray "
        data-test="container"
      >
        <div
          className="row"
          data-test="row"
        >
          <div
            className="col-md-12 col-lg-8 col-xl-8"
            data-test="col"
          >
            <div
              aria-label="carousel"
              className="carousel carousel-fade"
              data-test="carousel"
              length={4}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="carousel-inner"
                data-test="carousel-inner"
              >
                <div>
                  <div
                    className="carousel-item active carousel-slide-item 
            active-card
             mt-md-4 mb-md-4 pl-lg-3"
                    data-test="carousel-item"
                    style={
                      Object {
                        "left": "0",
                      }
                    }
                  >
                    <div
                      className="card-cascade wider card w-30 shadow-none Card"
                      data-test="card"
                    >
                      <div
                        className="container w-full"
                        data-test="container"
                      >
                        <div
                          className="row p-0"
                          data-test="row"
                        >
                          <div
                            className="col-12 col-md-4 col-lg-5 p-0"
                            data-test="col"
                          >
                            <img
                              alt="cover book"
                              className="imageCardCarsouel"
                              src="http://1.bp.blogspot.com/-yQklGusIOBE/Ul-oQ0_9vOI/AAAAAAAAfI4/XQTmNLaJZLQ/s1600/book-cover-poster-27.jpg"
                            />
                          </div>
                          <div
                            className="col-md-1 col-lg-1"
                            data-test="col"
                          />
                          <div
                            className="col-12 col-md-7 col-lg-6 mb-0 mt-auto"
                            data-test="col"
                          >
                            <div
                              className="card-body p-lg-0 text-center text-md-left text-lg-left mx-auto mt-md-2"
                              data-test="card-body"
                            >
                              <div>
                                <h1
                                  className="fontHeader1 mb-0 font-weight-normal  text-2xl"
                                >
                                  Alice Mayer
                                </h1>
                              </div>
                              <p
                                className="card-text"
                                data-test="card-text"
                              >
                                <p
                                  className="mb-1 fontHeader2 font-normal font-weight-normal "
                                >
                                  The Shadows: A Novel
                                </p>
                                <p
                                  className="mb-1 parg1 font-weight-normal"
                                >
                                  formate:Hardback
                                </p>
                                <p
                                  className="font-black fontPrice font-weight-normal "
                                >
                                  $31.5
                                </p>
                                <div
                                  className="mb-3"
                                >
                                  <div
                                    className="inline-block mr-2"
                                  >
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1 font-light"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1 font-light"
                                    />
                                      
                                    22,333
                                     views
                                  </div>
                                </div>
                              </p>
                              <button
                                className="buttomText text-center mycolorPink w-5/6 shadow-xl  hover:shadow-lg h-10 text-black font-bold leading-3 rounded-full"
                              >
                                carsouelSection.addtocart
                              </button>
                            </div>
                            <div
                              className="h-16 relative"
                            >
                              <div
                                className="absolute bottom-0 right-0"
                              >
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-left fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-right fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="carousel-item active carousel-slide-item 
            inactive-card
             mt-md-4 mb-md-4 pl-lg-3"
                    data-test="carousel-item"
                    style={
                      Object {
                        "left": "100%",
                        "position": "absolute",
                      }
                    }
                  >
                    <div
                      className="card-cascade wider card w-30 shadow-none Card"
                      data-test="card"
                    >
                      <div
                        className="container w-full"
                        data-test="container"
                      >
                        <div
                          className="row p-0"
                          data-test="row"
                        >
                          <div
                            className="col-12 col-md-4 col-lg-5 p-0"
                            data-test="col"
                          >
                            <img
                              alt="cover book"
                              className="imageCardCarsouel"
                              src="http://www.snugglyoranges.com/wp-content/uploads/2014/09/Unspoken.jpg"
                            />
                          </div>
                          <div
                            className="col-md-1 col-lg-1"
                            data-test="col"
                          />
                          <div
                            className="col-12 col-md-7 col-lg-6 mb-0 mt-auto"
                            data-test="col"
                          >
                            <div
                              className="card-body p-lg-0 text-center text-md-left text-lg-left mx-auto mt-md-2"
                              data-test="card-body"
                            >
                              <div>
                                <h1
                                  className="fontHeader1 mb-0 font-weight-normal  text-2xl"
                                >
                                  Alice Mayer
                                </h1>
                              </div>
                              <p
                                className="card-text"
                                data-test="card-text"
                              >
                                <p
                                  className="mb-1 fontHeader2 font-normal font-weight-normal "
                                >
                                  The Shadows: A Novel
                                </p>
                                <p
                                  className="mb-1 parg1 font-weight-normal"
                                >
                                  formate:Hardback
                                </p>
                                <p
                                  className="font-black fontPrice font-weight-normal "
                                >
                                  $31.5
                                </p>
                                <div
                                  className="mb-3"
                                >
                                  <div
                                    className="inline-block mr-2"
                                  >
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fas fa-star-half-alt text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1 font-light"
                                    />
                                      
                                    22k
                                     views
                                  </div>
                                </div>
                              </p>
                              <button
                                className="buttomText text-center mycolorPink w-5/6 shadow-xl  hover:shadow-lg h-10 text-black font-bold leading-3 rounded-full"
                              >
                                carsouelSection.addtocart
                              </button>
                            </div>
                            <div
                              className="h-16 relative"
                            >
                              <div
                                className="absolute bottom-0 right-0"
                              >
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-left fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-right fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="carousel-item active carousel-slide-item 
            inactive-card
             mt-md-4 mb-md-4 pl-lg-3"
                    data-test="carousel-item"
                    style={
                      Object {
                        "left": "100%",
                        "position": "absolute",
                      }
                    }
                  >
                    <div
                      className="card-cascade wider card w-30 shadow-none Card"
                      data-test="card"
                    >
                      <div
                        className="container w-full"
                        data-test="container"
                      >
                        <div
                          className="row p-0"
                          data-test="row"
                        >
                          <div
                            className="col-12 col-md-4 col-lg-5 p-0"
                            data-test="col"
                          >
                            <img
                              alt="cover book"
                              className="imageCardCarsouel"
                              src="https://s-media-cache-ak0.pinimg.com/736x/78/c5/f0/78c5f0de85565a0fd4a911f2681f56e3--book-cover-art-cover-books.jpg"
                            />
                          </div>
                          <div
                            className="col-md-1 col-lg-1"
                            data-test="col"
                          />
                          <div
                            className="col-12 col-md-7 col-lg-6 mb-0 mt-auto"
                            data-test="col"
                          >
                            <div
                              className="card-body p-lg-0 text-center text-md-left text-lg-left mx-auto mt-md-2"
                              data-test="card-body"
                            >
                              <div>
                                <h1
                                  className="fontHeader1 mb-0 font-weight-normal  text-2xl"
                                >
                                  Alice Mayer
                                </h1>
                              </div>
                              <p
                                className="card-text"
                                data-test="card-text"
                              >
                                <p
                                  className="mb-1 fontHeader2 font-normal font-weight-normal "
                                >
                                  The Shadows: A Novel
                                </p>
                                <p
                                  className="mb-1 parg1 font-weight-normal"
                                >
                                  formate:Hardback
                                </p>
                                <p
                                  className="font-black fontPrice font-weight-normal "
                                >
                                  $31.5
                                </p>
                                <div
                                  className="mb-3"
                                >
                                  <div
                                    className="inline-block mr-2"
                                  >
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fas fa-star-half-alt text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1 font-light"
                                    />
                                      
                                    22k
                                     views
                                  </div>
                                </div>
                              </p>
                              <button
                                className="buttomText text-center mycolorPink w-5/6 shadow-xl  hover:shadow-lg h-10 text-black font-bold leading-3 rounded-full"
                              >
                                carsouelSection.addtocart
                              </button>
                            </div>
                            <div
                              className="h-16 relative"
                            >
                              <div
                                className="absolute bottom-0 right-0"
                              >
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-left fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-right fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="carousel-item active carousel-slide-item 
            inactive-card
             mt-md-4 mb-md-4 pl-lg-3"
                    data-test="carousel-item"
                    style={
                      Object {
                        "left": "100%",
                        "position": "absolute",
                      }
                    }
                  >
                    <div
                      className="card-cascade wider card w-30 shadow-none Card"
                      data-test="card"
                    >
                      <div
                        className="container w-full"
                        data-test="container"
                      >
                        <div
                          className="row p-0"
                          data-test="row"
                        >
                          <div
                            className="col-12 col-md-4 col-lg-5 p-0"
                            data-test="col"
                          >
                            <img
                              alt="cover book"
                              className="imageCardCarsouel"
                              src="https://th.bing.com/th/id/OIP.Zf2SkeIiSI4ymvT2cc4u2AHaKd?pid=Api&rs=1"
                            />
                          </div>
                          <div
                            className="col-md-1 col-lg-1"
                            data-test="col"
                          />
                          <div
                            className="col-12 col-md-7 col-lg-6 mb-0 mt-auto"
                            data-test="col"
                          >
                            <div
                              className="card-body p-lg-0 text-center text-md-left text-lg-left mx-auto mt-md-2"
                              data-test="card-body"
                            >
                              <div>
                                <h1
                                  className="fontHeader1 mb-0 font-weight-normal  text-2xl"
                                >
                                  Alice Mayer
                                </h1>
                              </div>
                              <p
                                className="card-text"
                                data-test="card-text"
                              >
                                <p
                                  className="mb-1 fontHeader2 font-normal font-weight-normal "
                                >
                                  The Shadows: A Novel
                                </p>
                                <p
                                  className="mb-1 parg1 font-weight-normal"
                                >
                                  formate:Hardback
                                </p>
                                <p
                                  className="font-black fontPrice font-weight-normal "
                                >
                                  $31.5
                                </p>
                                <div
                                  className="mb-3"
                                >
                                  <div
                                    className="inline-block mr-2"
                                  >
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1 font-light"
                                    />
                                    <icon
                                      className="fa fa-star text-yellow-300 mr-1 font-light"
                                    />
                                      
                                    22k
                                     views
                                  </div>
                                </div>
                              </p>
                              <button
                                className="buttomText text-center mycolorPink w-5/6 shadow-xl  hover:shadow-lg h-10 text-black font-bold leading-3 rounded-full"
                              >
                                carsouelSection.addtocart
                              </button>
                            </div>
                            <div
                              className="h-16 relative"
                            >
                              <div
                                className="absolute bottom-0 right-0"
                              >
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-left fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                                <button
                                  className="btn Ripple-parent shadow-sm p-0 h-8"
                                  data-test="button"
                                  onClick={[Function]}
                                  onMouseUp={[Function]}
                                  onTouchStart={[Function]}
                                  type="button"
                                >
                                  <i
                                    className="fa fa-chevron-right fa-lg px-2 py-1"
                                    data-test="fa"
                                  />
                                  <div
                                    className="Ripple "
                                    data-test="waves"
                                    style={
                                      Object {
                                        "height": "0px",
                                        "left": "0px",
                                        "top": "0px",
                                        "width": "0px",
                                      }
                                    }
                                  />
                                </button>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-xs-12 col-sm-12 col-md-12 col-lg-4 col-xl-4 mt-3 mb-md-0 mt-md-0 mb-lg-4 mt-lg-4  "
            data-test="col"
          >
            <div
              className="container px-md-4"
              data-test="container"
            >
              <div
                className="row justify-content-around"
                data-test="row"
              >
                <div
                  className="col-3 col-sm-3 col-md-3 col-lg-6 px-2 pl-md-1 py-1 "
                  data-test="col"
                >
                  <div
                    className="view"
                    data-test="view"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={Object {}}
                  >
                    <img
                      className="imgGrideCarsouel mx-lg-0 "
                      src="http://1.bp.blogspot.com/-yQklGusIOBE/Ul-oQ0_9vOI/AAAAAAAAfI4/XQTmNLaJZLQ/s1600/book-cover-poster-27.jpg"
                    />
                  </div>
                </div>
                <div
                  className="col-3 col-sm-3 col-md-3 col-lg-6 px-2 pl-md-1 py-1 "
                  data-test="col"
                >
                  <div
                    className="view"
                    data-test="view"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={Object {}}
                  >
                    <img
                      className="imgGrideCarsouel mx-lg-0 "
                      src="http://www.snugglyoranges.com/wp-content/uploads/2014/09/Unspoken.jpg"
                    />
                    <div
                      className="mask rgba-black-light"
                      data-test="mask"
                    />
                  </div>
                </div>
                <div
                  className="col-3 col-sm-3 col-md-3 col-lg-6 px-2 pl-md-1 py-1 "
                  data-test="col"
                >
                  <div
                    className="view"
                    data-test="view"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={Object {}}
                  >
                    <img
                      className="imgGrideCarsouel mx-lg-0 "
                      src="https://s-media-cache-ak0.pinimg.com/736x/78/c5/f0/78c5f0de85565a0fd4a911f2681f56e3--book-cover-art-cover-books.jpg"
                    />
                    <div
                      className="mask rgba-black-light"
                      data-test="mask"
                    />
                  </div>
                </div>
                <div
                  className="col-3 col-sm-3 col-md-3 col-lg-6 px-2 pl-md-1 py-1 "
                  data-test="col"
                >
                  <div
                    className="view"
                    data-test="view"
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    style={Object {}}
                  >
                    <img
                      className="imgGrideCarsouel mx-lg-0 "
                      src="https://th.bing.com/th/id/OIP.Zf2SkeIiSI4ymvT2cc4u2AHaKd?pid=Api&rs=1"
                    />
                    <div
                      className="mask rgba-black-light"
                      data-test="mask"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
